src/GafferCycles/IECoreCyclesPreview/ShaderNetworkAlgo.cpp

```diff
@@ -773,3 +775,354 @@ IECoreScene::ShaderNetworkPtr convertLightShader( const IECoreScene::ShaderNetwo
+{
+	shader->parameters()[name] = new typename DataTraits<T>::DataType( parameterValue( usdShader, usdName, defaultValue ) );
+
+	if( ShaderNetwork::Parameter input = network->input( { shaderHandle, usdName } ) )
+	{
+		if( name != usdName )
+		{
+			network->addConnection( { input, { shaderHandle, name } } );
+			network->removeConnection( { input, { shaderHandle, usdName } } );
+		}
+	}
+}
+
+const InternedString g_angleParameter( "angle" );
+const InternedString g_castShadowParameter( "cast_shadow" );
+const InternedString g_colorParameter( "color" );
+const InternedString g_colorTemperatureParameter( "colorTemperature" );
+const InternedString g_enableColorTemperatureParameter( "enableColorTemperature" );
+const InternedString g_exposureParameter( "exposure" );
+const InternedString g_filenameParameter( "filename" );
+const InternedString g_heightParameter( "height" );
+const InternedString g_intensityParameter( "intensity" );
+const InternedString g_lengthParameter( "length" );
+const InternedString g_mathTypeParameter( "math_type" );
+const InternedString g_normalizeParameter( "normalize" );
+const InternedString g_parametricParameter( "parametric" );
+const InternedString g_positionParameter( "position");
+const InternedString g_projectionParameter( "projection");
+const InternedString g_radiusParameter( "radius" );
+const InternedString g_rotationParameter( "rotation" );
+const InternedString g_shadowEnableParameter( "shadow:enable" );
+const InternedString g_shapingConeAngleParameter( "shaping:cone:angle" );
+const InternedString g_shapingConeSoftnessParameter( "shaping:cone:softness" );
+const InternedString g_sizeParameter( "size" );
+const InternedString g_spotAngleParameter( "spot_angle" );
+const InternedString g_spotSmoothParameter( "spot_smooth" );
+const InternedString g_textureFileParameter( "texture:file" );
+const InternedString g_textureFormatParameter( "texture:format" );
+const InternedString g_texMappingScaleParameter( "tex_mapping__scale" );
+const InternedString g_texMappingYMappingParameter( "tex_mapping__y_mapping" );
+const InternedString g_texMappingZMappingParameter( "tex_mapping__z_mapping" );
+const InternedString g_treatAsPointParameter( "treatAsPoint" );
+const InternedString g_vectorParameter( "vector" );
+const InternedString g_vector1Parameter( "vector1" );
+const InternedString g_vector2Parameter( "vector2" );
+const InternedString g_widthParameter( "width" );
+
+void transferUSDLightParameters( ShaderNetwork *network, InternedString shaderHandle, const Shader *usdShader, Shader *shader )
+{
+	Color3f color = parameterValue( usdShader, g_colorParameter, Color3f( 1 ) );
+	if( parameterValue( usdShader, g_enableColorTemperatureParameter, false ) )
+	{
+		color *= blackbody( parameterValue( usdShader, g_colorTemperatureParameter, 6500.0f ) );
```

Comment:
Yeah, definitely would be nice to get this right, but I would like to call it out of scope this time round. My feeling is that the StandardLightVisualiser is pretty hard to maintain already because it's trying to cater to all flavours of light everywhere through an increasingly esoteric bunch of metadata. I think the best approach might be to create a dedicated USDLightVisualiser that can drop all those permutations and just implement the USD semantics directly.